### PR TITLE
[operator] define opentelemetry-operator.image in helper tpl

### PR DESCRIPTION
**Pull request by JaredTan95**
As described in the otel-operator official [documentation](https://github.com/open-telemetry/opentelemetry-operator?tab=readme-ov-file#opentelemetry-operator-vs-kubernetes-vs-cert-manager), there are compatibility issues for different kubernetes versions.
> | OpenTelemetry Operator         | Kubernetes|
| ---------------------- | ----------------------- |
| v0.99.0 | v1.23 to v1.29 |
| v0.85.0 | v1.19 to v1.28 |

If we need to consider the operator will work properly under different kubernetes versions.
there is a one way is through `semverCompare` functions to compare kubernetes Version like `semverCompare ">= 1.23-0" .Capabilities.KubeVersion.Version ` and set the `image.tag=0.99.0`, if `semverCompare "< 1.23-0" .Capabilities.KubeVersion.Version ` set the `iamge.tag=0.85.0` .

When we use `otel-operator` as a sub-chart, the parent chart can override the logical of `opentelemetry-operator.image` to implement the above logic. 

*So this PR just provides an entry to a custom `opentelemetry-operator.image`.*

### CODE REVIEW

File: charts/opentelemetry-operator/templates/_helpers.tpl [hunk 1]
```diff
@@ -142,3 +142,10 @@ Return the name of the cert-manager.io/inject-ca-from annotation for webhooks an
 {{- printf "%s/%s" .Release.Namespace (include "opentelemetry-operator.webhookCertName" .) }}
 {{- end }}
 {{- end }}
+
+{{/*
+The image to use for opentelemetry-operator.
+*/}}
+{{- define "opentelemetry-operator.image" -}}
+{{- printf "%s:%s" .Values.manager.image.repository (default .Chart.AppVersion .Values.manager.image.tag) }}
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/templates/_helpers.tpl:150]: I am missing what this template is doing differently that the existing logic

Comment (JaredTan95) [reply charts/opentelemetry-operator/templates/_helpers.tpl:150]: there is no difference

Comment (TylerHelmuth) [reply charts/opentelemetry-operator/templates/_helpers.tpl:150]: But how does this change help the issue described in the PR? Also, our chart explicitly does not support `1.23`, so I don't really want to add features for that version.

**Comment by JaredTan95** [comment]
I'm sorry, my description above is a little rough.

@TylerHelmuth  I also agree that we shoul not do multi-version(1.23) compatibility in the community upstream. But, the downstream can impl multi-version compatibility in the following ways:

1. use `otel-operator` as sub-chart:

```bash
dependencies:
  - name: opentelemetry-operator
    version: 0.58.0
    repository: https://open-telemetry.github.io/opentelemetry-helm-charts
```

2. then they can override `{{- define "opentelemetry-operator.image" -}}` logic in there own parent template(`_helpers.tpl`) to override the value of the operator image:

```bash
{{- define "opentelemetry-operator.image" -}}
{{- $suggest_tag := .Chart.AppVersion }}
  {{- if (semverCompare "< 1.23-0" .Capabilities.KubeVersion.Version) }}
    {{- $suggest_tag = 0.85.0 }}
  {{- else }}
    {{- $suggest_tag = 0.99.0 }}
  {{- end }}

{{- printf "%s:%s" .Values.manager.image.repository $suggest_tag }}
{{- end }}
```

File: charts/opentelemetry-operator/templates/_helpers.tpl (hunk 1 above)
Comment (JaredTan95) [reply charts/opentelemetry-operator/templates/_helpers.tpl:150]: I'm sorry, my description above is a little rough.

@TylerHelmuth  I also agree that we shoul not do multi-version(1.23) compatibility in the community upstream. But, the downstream can impl multi-version compatibility in the following ways:

1. use `otel-operator` as sub-chart:

```bash
dependencies:
  - name: opentelemetry-operator
    version: 0.58.0
    repository: https://open-telemetry.github.io/opentelemetry-helm-charts
```

2. then they can override `{{- define "opentelemetry-operator.image" -}}` logic in there own parent template(`_helpers.tpl`) to override the value of the operator image:

```bash
{{- define "opentelemetry-operator.image" -}}
{{- $suggest_tag := .Chart.AppVersion }}
  {{- if (semverCompare "< 1.23-0" .Capabilities.KubeVersion.Version) }}
    {{- $suggest_tag = 0.85.0 }}
  {{- else }}
    {{- $suggest_tag = 0.99.0 }}
  {{- end }}

{{- printf "%s:%s" .Values.opentelemetry-operator.manager.image.repository $suggest_tag }}
{{- end }}
```

Comment (TylerHelmuth) [reply charts/opentelemetry-operator/templates/_helpers.tpl:150]: >  override {{- define "opentelemetry-operator.image" -}}
 woah, I actually didn't know this was possible.

**Review by TylerHelmuth — APPROVED** [review]

**Review by Allex1 — APPROVED** [review]
@JaredTan95 pls bump the chart amd generate examples

**Comment by JaredTan95** [comment]
> @JaredTan95 pls bump the chart amd generate examples

yes, another PR mergers caused

**Pull request merged**